docs/contracts/v4/concepts/02-flash-accounting.mdx

```diff
@@ -4,12 +4,97 @@ title: Flash Accounting
 
 # Flash Accounting
 
+In previous versions of Uniswap, every time a swap was made - including multi-hop swap - output tokens had to be transferred out the Pool contract and then transferred into whatever their destination was (next pool in case of multi-hop, or back to the user in case of simple swap).
+
+This design led to inefficiencies because moving tokens around by calling external functions on their smart contracts - especially in a multi-hop swap - was quite expensive. But also, this was the only design possible since each pool was it's own contract and we needed to move tokens in and out of them to keep the logic sound.
```

Comment:
```suggestion
This design incurred inefficiencies because transferring tokens with external calls to their smart contracts - especially in a multi-hop swap - is quite expensive. This design was required since each pool was its own contract and token transfers were required to maintain accounting and solvency
```